
# *Component variables

Filter: `^\w*Component$`
Allowed: `PascalCase` for variables ending with `Component`.

## ✅ Good

```ts
const EditActionComponent = () => null;
```

## ❌ Bad

```ts
const editActionComponent = () => null;
```
